### ISSUE TITLE
`#material` and `material=` methods should be undefined for guide objects

### ISSUE THREAD

**Issue by DanRathbun** [SketchUp/api-issue-tracker#785]
#### SketchUp Ruby API Issue

**`#material` and `#material=` methods _might_ be **undefined** for subclasses [`ConstructionLine`](https://ruby.sketchup.com/Sketchup/ConstructionLine.html) and [`ConstructionPoint`](https://ruby.sketchup.com/Sketchup/ConstructionPoint.html).**

These methods are inherited from [`Drawingelement`](https://ruby.sketchup.com/Sketchup/Drawingelement.html) but have no effect as Guide **color** is controlled by the style in use and accessed via `#rendering_options["ConstructionColor"]` for either the model or a scene page if it is using it's own style.

Basically `#material=` is a no op and `#material` always returns `nil`.

**At the very least** a note in the subclass introductions could explain this and mention `#rendering_options["ConstructionColor"]`.

Ref forum topic:  [Failed to set material for construction line](https://forums.sketchup.com/t/failed-to-set-material-for-construction-line/198446)

**Comment by DanRathbun** [comment]
(Musing) It makes me wonder how YARD would treat inherited methods that are undefined in a subclass.

**Comment by CAUL2000** [comment]
The optimal solution to this would be to actually display guides with materials. Currently this has to be accomplished by using color by tag which is less than ideal.

**Comment by DanRathbun** [comment]
@CAUL2000 You should likely open a separate feature request issue and reference this issue. (All it takes is including a pound sign and number in the text.)

I feel it would make no sense to have `#material` and `#material=` methods, as the [`Material`](https://ruby.sketchup.com/Sketchup/Material.html) class has features that could not apply to guides.

I'd think rather that simple `#color` and `#color=` methods (like layer objects have) would be more appropriate.
If a guide object's `color` property is not set then it should use the `"ConstructionColor"` setting from the style in use.
I would also think (for backward compatibility) that there might be a `Style`/`RenderingOptions option that must be set for the renderer to use individual guide color properties. (Ie, the style global guide color is the current  default.)

In this new issue, **please** describe some scenarios why we'd use guides of differing colors.
